src/cockpit/superuser.py

```diff
@@ -99,9 +99,11 @@ class AuthorizeResponder(CockpitResponder):
     def __init__(self, router: Router):
         self.router = router
 
-    async def do_askpass(self, messages: str, prompt: str, hint: str) -> str:
+    async def do_askpass(self, messages: str, prompt: str, hint: str) -> 'str | None':
         hexuser = ''.join(f'{c:02x}' for c in getpass.getuser().encode('ascii'))
-        return await self.router.request_authorization(f'plain1:{hexuser}')
+        password = await self.router.request_authorization(f'plain1:{hexuser}')
+        # translate "no password" from authorize protocol (empty string) to ferny protocol (None)
+        return None if password == '' else password
```

Comment:
What happens when we return `None` here? Is the peer bridge killed?

Reply:
We can't kill sudo in the "send signal" sense, as it's running as root (so no permission). ferny simply closes its stdin, so like the equivalent of ^D on the terminal ("sudo: no password was provided"). That's good enough, and even killing wouldn't help -- at that point sudo already started the PAM conversation, so it'll be counted as failed login attempt either way.